### ISSUE TITLE
S1854 - False positive when using linq `collection.Aggregate`

### ISSUE THREAD

**Issue by FetzenRndy** [SonarSource/sonar-csharp#2095]
### Description

False-positive when using the linq function `.Aggregate`

### Reproduction steps

This code will trigger rule `S1854` but it should not

```csharp
return collection.Aggregate((decimal) 0, (acc, item) => acc += item.Value);
```

The warning is:
`Warning	S1854	Remove this useless assignment to local variable 'acc'.`

### Expected behavior

Do not suggest removing the variable.

### Actual behavior

It does suggest removing the variable.

### Known workarounds

- None (except of course disabling the rule)

### Related information

* SonarC# - Latest
* Visual Studio - 2017 Latest

**Comment by valhristov** [comment]
The assignment is actually not needed. It is the return value of the function that replaces the accumulator, not the value of the modified argument (actually the argument is not returned back because it is not a reference).

For example, the following code generates the same result:

    var list = new[] { 1, 2, 3, 4 };
    var result1 = list.Aggregate(0, (acc, item) => acc + item);
    Console.WriteLine(result1); // 10
    var result2 = list.Aggregate(0, (acc, item) => acc += item);
    Console.WriteLine(result2); // 10
